plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/DeployJob.java

```diff
@@ -98,15 +99,17 @@ public DeployJob(IProject project, Credential credential, IPath workDirectory,
       ProcessOutputLineListener stagingStdoutLineListener,
       ProcessOutputLineListener stderrLineListener,
       DefaultDeployConfiguration deployConfiguration,
-      boolean includeOptionalConfigurationFiles) {
-    super(Messages.getString("deploy.standard.runnable.name")); //$NON-NLS-1$
+      boolean includeOptionalConfigurationFiles,
+      DeployEnvironmentDelegate environmentDelegate) {
```

Comment:
Would it make sense to move some of these arguments as getters and setters (e.g., includeOptionalConfigurationFiles).